protocol/messenger.go

```diff
@@ -917,9 +924,68 @@ func (m *Messenger) Start() (*MessengerResponse, error) {
 		}
 	}
 
+	if m.processBackedupMessages {
+		err = m.startBackupFetchingTracking(response)
+		if err != nil {
+			return nil, err
+		}
+	}
+
 	return response, nil
 }
 
+func (m *Messenger) startBackupFetchingTracking(response *MessengerResponse) error {
+	// Add an acivity center notification to show that we are fetching back up messages
+	notification := &ActivityCenterNotification{
+		ID:        types.FromHex(backupSyncingNotificationID), // TODO put this in a constant
+		Type:      ActivityCenterNotificationTypeBackupSyncingFetching,
+		Timestamp: m.getTimesource().GetCurrentTime(),
+		Read:      false,
+		Deleted:   false,
+		UpdatedAt: m.GetCurrentTimeInMillis(),
+	}
+	err := m.addActivityCenterNotification(response, notification, nil)
+
+	if err != nil {
+		return err
+	}
+
+	// Add a timeout to mark the backup syncing as failed after 1 minute and 30 seconds
+	time.AfterFunc(1*time.Minute+30*time.Second, func() {
+		if m.fetchingBackedUpDataCompleted {
+			// Nothing to do, the fetching has completed successfully
+			return
+		}
+		// Update the AC notification to the failure state
+		hexBytesIds := []types.HexBytes{}
+		hexBytesIds = append(hexBytesIds, types.FromHex(backupSyncingNotificationID))
+		notifications, err := m.persistence.GetActivityCenterNotificationsByID(hexBytesIds)
+		if err != nil {
+			m.logger.Error("failed to get activity center notification", zap.Error(err))
+		} else if len(notifications) == 1 {
```

Comment:
Early return would improve readability here.
```suggestion
		if err != nil {
			m.logger.Error("failed to get activity center notification", zap.Error(err))
            return nil // or return err?
		}
```